netadapter/netadapter.go

```diff
@@ -165,3 +167,58 @@ func (na *NetAdapter) Broadcast(connectionIDs []*id.ID, message wire.Message) er
 	}
 	return nil
 }
+
+// GetBestLocalAddress returns the most appropriate local address to use
+// for the given remote address.
+func (na *NetAdapter) GetBestLocalAddress() (*wire.NetAddress, error) {
+	//TODO(libp2p) Reimplement this, and check reachability to the other node
+	if len(config.ActiveConfig().ExternalIPs) > 0 {
+		host, portString, err := net.SplitHostPort(config.ActiveConfig().ExternalIPs[0])
+		if err != nil {
+			portString = config.ActiveConfig().NetParams().DefaultPort
+		}
+		portInt, err := strconv.Atoi(portString)
+		if err != nil {
+			return nil, err
+		}
+
+		ip := net.ParseIP(host)
+		if ip == nil {
+			hostAddrs, err := net.LookupHost(host)
+			if err != nil {
+				return nil, err
+			}
+			ip = net.ParseIP(hostAddrs[0])
+			if ip == nil {
+				return nil, errors.Errorf("Cannot resolve IP address for host '%s'", host)
+			}
+		}
+		return wire.NewNetAddressIPPort(ip, uint16(portInt), wire.SFNodeNetwork), nil
+
```

Comment:
This part and the next part are almost entirely identical. Please extract it to a function.
Also, there's a superfluous empty line here.